system-addon/content-src/components/Sections/Sections.jsx

```diff
@@ -190,6 +190,35 @@ Section.defaultProps = {
 export const SectionIntl = connect(state => ({Prefs: state.Prefs}))(injectIntl(Section));
 
 export class _Sections extends React.PureComponent {
+  componentDidMount() {
+    this.props.window.addEventListener("beforeunload", this._gcGBlobUrls);
```

Comment:
Out of curiosity, why do we need to use beforeunload and not just unload?

beforeunload gives web content an opportunity to block a tab close, which means that when a page (even about:newtab) sets a beforeunload event handler, we have to bypass a bunch of shortcuts and do messaging to see if a tab wants to block tab closing.

So could we swap this beforeunload out for something else?

Reply:
Good catch. I just read up more about `beforeunload`, and I wasn't aware that it is a cancelable event. Thinking about it again, I think `unload` would be the right choice especially for the call in `Card.jsx`:

```javascript
this.props.window.addEventListener("beforeunload", this.removeCardImageRef); // bad
```

We only want to remove the image reference only if the window is guaranteed to be unloaded. Doing it in `beforeunload` creates a situation in which the reference count will be reduced by 1 even in the case when the tab cancels the unload event, and this violates the invariant that we have before (i.e. reference count refers to the number of images that are *currently* referencing a specific blob URL).

Yes, we should swap this with `unload`.